### DISCUSSION TITLE
Question about genetic algorithms

### DISCUSSION THREAD

**Discussion by georgekasa** [microsoft/AI-For-Beginners#284] [Q&A]
Hello, 
thanks to your small tutorial I started to learn genetic algorithms, I would like if you could comment or refer any books or websites which they dive more in genetic algorithms with examples ideally just like yours.

thank you in advance!

**Comment by hassimalik** [comment]
# Glad the tutorial helped!

- For going deeper into genetic algorithms, I’d recommend **“Practical Genetic Algorithms” by Randy L. Haupt and Sue Ellen Haupt**. It focuses quite a bit on practical applications and examples rather than only theory, so it may be close to what you’re looking for.

- Another good book is **“An Introduction to Genetic Algorithms” by Melanie Mitchell**. It’s a bit more introductory, but it explains the concepts clearly and includes exercises and implementation-oriented material.

## For a more hands-on path, I’d start with Mitchell for the fundamentals, then move to Haupt & Haupt for more practical applications.

# Hope that helps, and good luck with your GA experiments!
